tests/spec/object/spec-flatten.js

```diff
@@ -0,0 +1,42 @@
+define(['mout/object/flatten'], function (flatten) {
+
+    describe('object/flatten()', function () {
+
+        it('should recursively flatten the object', function () {
+            var obj = { a: 1, b: 2, c: { d: 3, e: { f: 4 } } },
+                result;
+
+            result = flatten(obj);
+            expect(result).toEqual({ a: 1, b: 2, 'c.d': 3, 'c.e.f': 4 });
+        });
+
+        it('should only flatten one layer if level is 1', function () {
+            var obj = { a: 1, b: 2, c: { d: 3, e: { f: 4 } } },
+                result;
+
+            result = flatten(obj, 1);
+            expect(result).toEqual({ a: 1, b: 2, 'c.d': 3, 'c.e': { f: 4 } });
+        });
+
+        it('should only flatten 2 layers if level is 2', function () {
+            var obj = { a: 1, b: 2, c: { d: 3, e: { f: 4, g: { h: 5 } } } },
+                result;
+
+            result = flatten(obj, 2);
+            expect(result).toEqual({ a: 1, b: 2, 'c.d': 3, 'c.e.f': 4, 'c.e.g': { h: 5 } });
```

Comment:
Seems kind of odd to assume props would be handled as `c[dot]e[dot]f`. Is there another implementation this is based off of?